### PR TITLE
Remove pin_subpackage

### PR DESCRIPTION
**Pull request by hobu**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hobu** [comment]
Removing this because pins for PDAL are now in conda-forge-pinning

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,10 +9,8 @@ source:
   sha256: f6ae0f3dc012b19c70dde2361799ecac0cdcbeb9cad5cfd96313c9fdc8608f32
 
 build:
-  number: 1
+  number: 2
   skip: true  # [ppc64le]
-  run_exports:
-    - {{ pin_subpackage('pdal', max_pin='x.x') }}
```
Comment (isuruf) [inline recipe/meta.yaml:15]: This should be there. The pin in `conda-forge-pinning` affects what we build against (`host`) and this dictates what we run with (`run`) when we include pdal in host.

Comment (hobu) [reply recipe/meta.yaml:15]: whoops!

Comment (hobu) [reply recipe/meta.yaml:15]: reverted in #259

**Pull request merged**